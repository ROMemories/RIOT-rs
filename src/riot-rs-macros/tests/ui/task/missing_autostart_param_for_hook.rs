#![no_main]
#![feature(impl_trait_in_assoc_type)]
#![feature(type_alias_impl_trait)]
#![feature(used_with_arg)]

// As the macro will fail, this import will not get used
#[allow(unused_imports)]
use riot_rs::embassy::usb::UsbBuilderHook;

// FAIL: using hooks require the task to be autostart
#[riot_rs::task(usb_builder_hook)]
async fn main() {}
