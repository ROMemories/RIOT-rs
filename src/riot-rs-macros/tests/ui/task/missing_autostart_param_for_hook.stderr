error: custom attribute panicked
  --> tests/ui/task/missing_autostart_param_for_hook.rs:10:1
   |
10 | #[riot_rs::task(usb_builder_hook)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: the task must be `autostart` to instantiate hooks
